q2_diversity/_filter.py

```diff
@@ -25,3 +27,18 @@ def filter_distance_matrix(distance_matrix: skbio.DistanceMatrix,
     except skbio.stats.distance.DissimilarityMatrixError:
         raise ValueError(
             "All samples were filtered out of the distance matrix.")
+
+
+# This function filters the SampleData[AlphaDiversity] table by the metadata
+# only sample present in the metadata will remain
+def filter_alpha_diversity_artifact(alpha_diversity: pd.Series,
+                                    metadata: qiime2.Metadata,
+                                    exclude_ids: bool = False) -> pd.Series:
+    ids_to_keep = metadata.get_ids()
+    if exclude_ids:
+        ids_to_keep = set(alpha_diversity.index) - set(ids_to_keep)
+    filtered_table = alpha_diversity[alpha_diversity.index.isin(ids_to_keep)]
```

Comment:
I don't think "table" is technically correct (`pd.Series` is one-dimensional`), so maybe `filtered_metrics` or something